cloud/ec2_price.go

```diff
@@ -339,19 +339,23 @@ func (m *ec2Manager) getProvider(ctx context.Context, h *host.Host, ec2settings
 		spotPrice     float64
 		az            string
 	)
-	if h.UserHost || m.provider == onDemandProvider || m.provider == autoProvider {
-		onDemandPrice, err = pkgCachingPriceFetcher.getEC2OnDemandCost(ctx, m.client, getOsName(h), ec2settings.InstanceType, ec2settings.getRegion())
-		if err != nil {
-			return 0, errors.Wrap(err, "error getting ec2 on-demand cost")
+	// price fetcher tool only used for the default region
+	if ec2settings.getRegion() == evergreen.DefaultEC2Region {
```

Comment:
I don't really understand the context behind this change but aren't we already getting the on demand price from the requested region? So why do we need to do this check?

Reply:
This is for multi-region support: most regions don't support these operations, so for the case of other regions we need to skip this block.

Reply:
What's the difference between one region and another though? I would think that all regions have to support getting prices for their products

Reply:
I don't make the rules haha, this price list is probably pretty new, it's only for us-east-1 and ap-south-1
https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/using-pelong.html

Reply:
aws 🤦